### DISCUSSION TITLE
Using rgb(x,y,z) color for icon_color

### DISCUSSION THREAD

**Discussion by GizmoGoody** [piitaya/lovelace-mushroom#507] [Q&A] [answered]
More context here: https://community.home-assistant.io/t/using-rgb-x-y-z-color-for-icon-color/422775/12

Is it possible to use rgb values to set the icon color in Mushroom cards (e.g. Template Card)?

**Comment by seraphx2** [comment]
I currently came across a situation I would like this ability as well.
Hue lights/light groups report an rgb value that consists of the rgb values of all lights in the group "averaged" out to an attribute in the format `rgb_color: 255, 254, 250`.
It would be cool to just be able to use that value and make the icon reflect that color, just like the option to "use entity color" concept on dedicated light cards.

**Comment by piitaya** [comment, marked as answer]
Hi 👋
RGB color format is not available yet. May be it can be added but that's not the priority because it's a very advanced usage (user needs to manipulate RGB value formula inside template 🤯).
For now, as a workaround, you can convert it to the hexa format like I suggested it in [this issue](https://github.com/piitaya/lovelace-mushroom/issues/343). It works well 🙂

**Comment by seraphx2** [comment]
Based on what piitaya said, I was able to do a better search on Google and found this way of getting the values converted:
```
{% if is_state('light.living_room', 'on') %}
  {{ '#%02x%02x%02x' % state_attr('light.living_room','rgb_color') }}
{% endif %}
```

This works better with the `rgb_color` attribute that lights tend to have with the 3 values already comma delimited.

I have no clue how this works, but it does and that's good enough for me  lol

**Reply by GizmoGoody** [reply]
That syntax is also toward the bottom of the link in my first post. I agree...whatever works!

**Reply by seraphx2** [reply]
lmao. that's exactly the thread where I got it from after searching on Google. didn't even realize it was you. haha

**Reply by GizmoGoody** [reply]
Ha! Goes to show us that sometimes we have to ask these questions to even make these answers available on the interwebs!

Here's the syntax I've started using for simple if/then statements:

```
icon_color: >
  {{ iif(is_state('light.tripod_state','on'), "'#%02x%02x%02x' % state_attr('light.tripod','rgb_color')", 'grey') }}
```

It's more compact and actually similar to how I would write an if/then statement in Excel...which is a program I already understand. ;-P

**Reply by piitaya** [reply]
You can use the `disabled` color instead of` grey` 🙂

**Reply by GizmoGoody** [reply]
Thanks. That is a little lighter. I'm thankful for find and replace ;-)
